controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -198,10 +188,57 @@ public void cancelChange(ApplicationId applicationId, boolean keepApplicationCha
         });
     }
 
+    /** Returns the set of all jobs which have changes to propagate from the upstream steps, sorted by job. */
+    public Stream<Job> computeReadyJobs() {
```

Comment:
Since this is `public` I suggest returning `List`. Returning `Stream` is a bit icky as the result of the lazy operations (e.g. `this::computeReadyJobs`) may be different at the time the caller `collect`s the stream.

Reply:
Yes, you're right.